modules/apps/redirect/redirect-service/src/main/java/com/liferay/redirect/internal/upgrade/v3_0_2/RedirectEntrySourceURLUpgradeProcess.java

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.redirect.internal.upgrade.v3_0_2;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.dao.jdbc.AutoBatchPreparedStatementUtil;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * @author Attila Bakay
+ */
+public class RedirectEntrySourceURLUpgradeProcess extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		try (PreparedStatement preparedStatement1 = connection.prepareStatement(
+				"select redirectEntryId, sourceURL from RedirectEntry");
+			PreparedStatement preparedStatement2 =
+				AutoBatchPreparedStatementUtil.concurrentAutoBatch(
+					connection,
+					"update RedirectEntry set sourceURL = ? where " +
+						"redirectEntryId = ?")) {
+
+			ResultSet resultSet = preparedStatement1.executeQuery();
+
+			while (resultSet.next()) {
+				String sourceURL = resultSet.getString(2);
+
+				String lowerCaseSourceURL = StringUtil.toLowerCase(sourceURL);
+
+				if (sourceURL.equals(
+						StringUtil.toLowerCase(lowerCaseSourceURL))) {
```

Comment:
`lowerCaseSourceURL` is already lowercase, there's no need to call `toLowerCase` here.

Reply:
That was a copy past issue after cleaning up my fix originally I used StringUtil.toLowerCase(sourceURL); everywhere and later optimized it by moving it to a new variable I forget to remove the StringUtil.tolower here.